ArduPlane/ArduPlane.cpp

```diff
@@ -226,24 +226,29 @@ void Plane::update_compass(void)
 /*
   do 10Hz logging
  */
-void Plane::update_logging1(void)
+void Plane::update_logging10(void)
 {
-    if (should_log(MASK_LOG_ATTITUDE_MED) && !should_log(MASK_LOG_ATTITUDE_FAST)) {
+    bool log_faster = (should_log(MASK_LOG_ATTITUDE_FULLRATE) || should_log(MASK_LOG_ATTITUDE_FAST));
+    if (should_log(MASK_LOG_ATTITUDE_MED) && !log_faster) {
         Log_Write_Attitude();
-    }
-
-    if (should_log(MASK_LOG_ATTITUDE_MED) && !should_log(MASK_LOG_IMU))
-        AP::ins().Write_IMU();
-
-    if (should_log(MASK_LOG_ATTITUDE_MED))
         ahrs.Write_AOA_SSA();
```

Comment:
Logging AOA only on `ATTITUDE_MED` is a little odd. Existing issue tho. I would be be tempted to leave it as was originally in its own if `should_log(MASK_LOG_ATTITUDE_MED)`. The else if makes it a little confusing.

Reply:
Tom wanted it logged whenever ATTITUDE was logged at least at 10Hz...